tests/test_stdin_encoding_1426.py

```
@@ -0,0 +1,713 @@
+) -> None:
+    """Undecodable password bytes must abort, not derive a key from a guess.
+
+    The pin is strict, so bytes that are not UTF-8 raise at the read
+    rather than being substituted. `_read_password` returns None, and the
+    caller already treats that as "aborted: empty or non-matching
+    password" and exits 1. Refusing is the right failure: the alternative
+    is writing an archive against a key nobody can reproduce.
+    """
+    env = dict(os.environ)
+    env.update(CP1252)
+    result = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)